src/map/roe.cpp

```diff
@@ -266,23 +273,25 @@ namespace roeutils
         return PChar->m_eminenceLog.complete[page] & (1 << bit);
     }
 
-    uint16 GetNumEminenceCompleted(CCharEntity* PChar)
-    {
-        uint16 completedCount = 0;
+    uint16 GetNumEminenceCompleted(CCharEntity* PChar){
+        uint16 completedCount {0};
 
-        for (int page = 0; page < 512; page++)
+        for (uint16 page = 0; page < 512; page++)
         {
-            int pageVal = PChar->m_eminenceLog.complete[page];
-
-            while (pageVal)
+            unsigned long bitIndex {0};
+            uint8 pageVal = PChar->m_eminenceLog.complete[page];
+            // Strip off and check only the set bits - Hidden records are not counted.
+            for ( ; pageVal; completedCount += !RoeSystem.HiddenRecords.test(page * 8 + bitIndex))
             {
-                completedCount += pageVal & 1;
-                pageVal >>= 1;
+                #ifdef _MSC_VER
+                    _BitScanForward(&bitIndex, pageVal);
+                #else
+                    bitIndex = __builtin_ctz(pageVal);
+                #endif
+                pageVal &= (pageVal - 1);
```

Comment:
If the logic here is:
```
while pageVal is non-zero:
    scan across it and test each set bit
    accumulate result into completedCount
    then progress to the next page
```
Can we do that with a while loop instead of this hard to parse for loop?

If that simplifies how this looks, it may or may not be useful to extract these intrinsic out into a helper in `common`:
```cpp
// Scans from LSB to MSB
unsigned long ScanForFirstSetBit(uint8 value)
{
    unsigned long bitIndex = 0;
    #ifdef _MSC_VER
        _BitScanForward(&bitIndex, value);
    #else
        bitIndex = __builtin_ctz(value);
    #endif
    return bitIndex;
}
```

Reply:
Whatever makes the code easier for others to maintain and understand.

Intrinsics could be broken out if someone wishes to, but I feel their use cases might be relatively niche.

Reply:
To follow up on this, I would argue that we should absolutely keep things like this in an isolated place with pre-defined wrappers.  This is not a question of niche cases, but more an issue of maintainability.  Its much easier to handle environment differences in a single location, than potentially peppering the code with one-off cases.